### PR TITLE
feat(azure-apps): Add azure-apps for azure related components

### PR DESCRIPTION
**Pull request by isantospardo**
# Description

- Creates an *azure-apps* subchart where to store specific Azure applications
- Adds secrets-store-csi-driver-provider-azure for vault

# Checklist

* [x] I updated the version in Chart.yaml
* [x] I updated applicable README.md files using  `pre-commit run -a`
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [x] If I updated a dependency tool, or app, this PR contains a short summary of the changes I'm pulling
* [x] CI is currently green and this is ready for review
* [ ] I am ready to test changes after they are applied and released

### CODE REVIEW

File: charts/azure-apps/examples/secrets-store-csi-driver-provider-azure.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+secretsStoreCsiDriverAzure:
+  enabled: true
+  project: infra-secrets-store-csi-azure
+  values:
+    # Install RBAC roles and bindings required for K8S Secrets syncing if true
+    syncSecret:
```
Comment (eyenx) [inline charts/azure-apps/examples/secrets-store-csi-driver-provider-azure.yaml:6]: This stuff should be under `secrets-store-csi-driver` as shown [here](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/charts/csi-secrets-store-provider-azure/values.yaml#L98)

File: charts/azure-apps/values.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# secretsStoreCsiDriverAzure -- [secrets-store-csi-driver-provider-azure](https://github.com/Azure/secrets-store-csi-driver-provider-azure) ([example](./examples/secrets-store-csi-driver-provider-azure.yaml))
+# @default -- -
+secretsStoreCsiDriverAzure:
+  # secretsStoreCsiDriverAzure.enabled -- Enable secrets-store-csi-driver-provider-azure
+  enabled: false
+  name: secrets-store-csi-driver-provider-azure
```
Comment (eyenx) [inline charts/azure-apps/values.yaml:6]: perhaps we should rather go for a naming concept like vault did:

https://github.com/hashicorp/vault-csi-provider

so: `azure-key-vault-csi-provider`  or `azure-kv-csi-provider` 

What are your thoughts? Also on the Namespace and Project name.

**Review by eyenx — CHANGES_REQUESTED** [review]

File: charts/azure-apps/values.yaml (hunk 2 above)
Comment (isantospardo) [reply charts/azure-apps/values.yaml:6]: Yes, I think the same, also, the shortest the better, so we can proceed with `azure-kv-csi-provider`. I will update, thanks for the suggestion.

File: charts/azure-apps/examples/secrets-store-csi-driver-provider-azure.yaml (hunk 1 above)
Comment (isantospardo) [reply charts/azure-apps/examples/secrets-store-csi-driver-provider-azure.yaml]: Missed it, thanks!

**Comment by eyenx** [comment]
I would also consider renaming the app name and all the files in examples if we are going with `azure-kv-csi-provider` probably use `azureKvCsiProvider` instead of the whole `SecretsStoreCsi...`

**Review by eyenx — CHANGES_REQUESTED** [review]
see above

File: charts/azure-apps/examples/secrets-store-csi-driver-provider-azure.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+azureKvCsiProvider:
```
Comment (eyenx) [inline charts/azure-apps/examples/secrets-store-csi-driver-provider-azure.yaml:1]: should we also rename this files to match `azure-kv-csi-provider` ?

**Review by eyenx — CHANGES_REQUESTED** [review]

File: charts/azure-apps/examples/secrets-store-csi-driver-provider-azure.yaml (hunk 3 above)
Comment (isantospardo) [reply charts/azure-apps/examples/secrets-store-csi-driver-provider-azure.yaml]: Yes, why not, to be consistent, thanks for the suggestion

**Review by eyenx — APPROVED** [review]

**Review by hairmare — COMMENTED** [review]
i concur with toni as in the the naming should probably match the repo name more closely

File: charts/azure-apps/templates/NOTES.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+The following apps are available:
+{{ if .Values.azureKvCsiProvider.enabled }}
+* azureKvCsiProvider
```
[the remark this answers is not in the document]
Comment (isantospardo) [reply charts/azure-apps/templates/NOTES.txt]: I changed the name to `secrets-store-csi-driver-provider-azure`, thanks for the suggestion.

**Pull request merged**